src/builtins.c

```diff
@@ -580,6 +580,14 @@ JL_CALLABLE(jl_f_throw)
     return jl_nothing;
 }
 
+JL_CALLABLE(jl_f_throw_methoderror)
+{
+    JL_NARGSV(throw_methoderror, 1);
+    size_t world = jl_get_tls_world_age();
+    jl_method_error((jl_function_t *)args[0], &args[1], nargs, world);
```

Comment:
```suggestion
    jl_method_error((jl_value_t *)args[0], &args[1], nargs, world);
```
since arbitrary object can be a functor?